GPflowOpt/design.py

```diff
@@ -67,3 +74,147 @@ def __init__(self, domain):
+       }
+
+    """
+
+    def __init__(self, size, domain, max_seed_size=None):
+        super(LatinHyperCube, self).__init__(size, domain)
+        self._max_seed_size = max_seed_size or domain.size
+
+    def generate(self):
+        # Generate several TPLHDs with growing seed, select the one with the best intersite distance
+        candidates = []
+        scores = []
+
+        for i in np.arange(1, min(self.size, self._max_seed_size) + 1):
+            if i < 3:
+                # Hardcoded seeds for 1 or two points.
+                seed = np.arange(1, i + 1)[:, None] * np.ones((1, self.domain.size))
+            else:
+                # Generate larger seeds recursively by creating small LHD's
+                seed = self._tplhs_design(i, np.ones((1, self.domain.size)))
+
+            # Create all designs and compute score
+            X = self._tplhs_design(self.size, seed)
+            candidates.append(X)
+            scores.append(np.min(pdist(candidates[-1])))
+
+        # Transform best design (highest score) to specified domain
+        cube = np.sum([ContinuousParameter('x{0}'.format(i), 1, self.size) for i in np.arange(self.domain.size)])
+        transform = cube >> self.domain
+        return np.clip(transform.forward(candidates[np.argmax(scores)]), self.domain.lower, self.domain.upper)
+
+    def _tplhs_design(self, npoints, seed):
+        """
+        Creates a LHD with the Translational propagation algorithm with specified seed and design size
+        :param npoints: size of design to generate (N), may differ from self.size.
+        :param seed: 2D ndarray, the seed to use. S x D
+        :return: LHD, 2D ndarray. N x D
+        """
+        ns, nv = seed.shape
+        nd = np.power(npoints / float(ns), 1 / float(nv))
+        ndStar = np.ceil(nd)
+
+        # Determine npStar, the amount of points we'll be generating
+        npStar = np.power(ndStar, nv) * ns if ndStar > nd else npoints
+
+        # First assert scale of seed, then generate
+        seed = self._rescale_seed(seed, npStar, ndStar)
+        X = self._create(seed, npStar, ndStar)
+
+        # In case the generated design is too big (as specified by npStar), get rid of some points
+        return self._resize(X, npoints)
+
+    @staticmethod
+    def _rescale_seed(seed, npStar, ndStar):
+        """
+        Rescales the seeding pattern
+        :param seed: 2D ndarray, S x D
+        :param npStar: size of the LHD to be generated. N* >= N
+        :param ndStar: number of translation steps for the seed in each dimension
+        :return: rescaled seed, 2D ndarray, S x D
+        """
+        ns, nv = seed.shape
+        if ns == 1:
+            seed = np.ones((1, nv))
+            return seed
+        uf = ns * np.ones(nv)
+        ut = ((npStar / ndStar) - ndStar * (nv - 1) + 1) * np.ones(nv)
+        a = (ut - 1) / (uf - 1)
+        b = ut - a * uf
+
+        return np.round(a * seed + b)
+
+    @staticmethod
+    def _create(seed, npStar, ndStar):
```

Comment:
_tplhs_design calls _create. but from the doc string it seems to do the same thing: create a LHD.

make the method name similar _tplhd_something and update the doc string a bit to make the distinction more clear?